kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/formatter/kotlinFormatter.kt

```diff
@@ -8,172 +8,182 @@ import com.intellij.formatting.FormatterImpl
 import com.intellij.formatting.Indent
 import com.intellij.formatting.Spacing
 import com.intellij.lang.ASTNode
+import com.intellij.lang.Language
 import com.intellij.openapi.editor.impl.DocumentImpl
 import com.intellij.openapi.util.TextRange
 import com.intellij.openapi.util.text.StringUtil
 import com.intellij.psi.codeStyle.CodeStyleSettings
+import com.intellij.psi.codeStyle.CommonCodeStyleSettings
 import com.intellij.psi.codeStyle.CommonCodeStyleSettings.IndentOptions
 import com.intellij.psi.formatter.FormatterUtil
 import com.intellij.util.text.CharArrayUtil
 import org.eclipse.core.resources.IFile
-import org.eclipse.core.resources.IProject
 import org.eclipse.jface.text.Document
 import org.eclipse.jface.text.IDocument
 import org.jetbrains.kotlin.core.model.getEnvironment
 import org.jetbrains.kotlin.eclipse.ui.utils.IndenterUtil
 import org.jetbrains.kotlin.eclipse.ui.utils.LineEndUtil
+import org.jetbrains.kotlin.idea.KotlinLanguage
+import org.jetbrains.kotlin.idea.formatter.KotlinCommonCodeStyleSettings
 import org.jetbrains.kotlin.idea.formatter.KotlinSpacingBuilderUtil
 import org.jetbrains.kotlin.idea.formatter.createSpacingBuilder
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.psi.KtPsiFactory
 import com.intellij.openapi.editor.Document as IdeaDocument
 
-@Volatile var settings: CodeStyleSettings = CodeStyleSettings(true)
+@Volatile
+var settings: CodeStyleSettings = object : CodeStyleSettings(true) {
+	override fun getCommonSettings(lang: Language): CommonCodeStyleSettings =
+			when (lang) {
+				KotlinLanguage.INSTANCE -> KotlinCommonCodeStyleSettings()
```

Comment:
Let's extract `KotlinCommonCodeStyleSettings()` somewhere to avoid creating it each time